File: sdks/go/examples/basic.go
package main

import (
	greplog "github.com/greplog/greplog-go/src"
)

func main() {
	// 1. Auto-capture only — captures panics (via greplog.Go) and HTTP metadata automatically
	greplog.Init()

	// 2. Service name override
	greplog.Init(greplog.Config{Service: "api-backend"})

	// Fail-open guarantee: these calls are safe even if the agent isn't running
	// 3. Manual log levels with structured details
	greplog.Error("payment failed", greplog.Details{"order_id": "ord-123", "amount": "99.99"})
	greplog.Warn("retrying request", greplog.Details{"attempt": "2"})
	greplog.Info("server started", greplog.Details{"port": "4000"})
	greplog.Debug("cache miss", greplog.Details{"key": "user:123"})
}

File: sdks/rust/examples/basic.rs
fn main() {
    // 1. Auto-capture only — captures panics and tracing events automatically
    greplog::init();

    // 2. Service name override
    greplog::init_with_config(greplog::Config {
        service: Some("api-backend".into()),
        ..Default::default()
    });

    // Fail-open guarantee: these calls are safe even if the agent isn't running
    // 3. Manual log levels with structured details
    greplog::error!("payment failed", &[("order_id", "ord-123"), ("amount", "99.99")]);
    greplog::warn!("retrying request", &[("attempt", "2")]);
    greplog::info!("server started", &[("port", "4000")]);
    greplog::debug!("cache miss", &[("key", "user:123")]);
}
